app/src/main/java/com/maasbodev/tasky/navigation/NavigationRoot.kt

```diff
@@ -0,0 +1,37 @@
+package com.maasbodev.tasky.navigation
+
+import androidx.compose.runtime.Composable
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import com.maasbodev.auth.presentation.login.LoginScreenRoot
+import com.maasbodev.auth.presentation.register.RegisterScreen
+
+@Composable
+fun NavigationRoot(
+	navController: NavHostController,
+	isLoggedIn: Boolean,
+) {
+	NavHost(
+		navController = navController,
+		startDestination = if (isLoggedIn) LoginNavObject else LoginNavObject
+	) {
+		composable<LoginNavObject> {
+			LoginScreenRoot(
+				onLoginSuccess = {
+					navController.navigate("run") {
+						popUpTo("auth") {
+							inclusive = true
+						}
+					}
+				},
+				onSignUpClick = {
+					navController.navigate(RegisterNavObject)
+				}
+			)
+		}
+		composable<RegisterNavObject> {
+			RegisterScreen()
+		}
```

Comment:
I like to work with one navigation graph per feature, since then you can conveniently pop the whole graph from the back stack when logging in for example without risking there are some screens left on the back stack.

Works just like defining routes:
```
@Serializable
data object AuthGraph

NavHost(...) {
    navigation<AuthGraph> {
        composable<LoginRoute> {
        
        }
        ...
    }
}
```

Reply:
This is a nested navigation, I know how to do it with the old jetpack compose navigation, but still could not find how to do it with the new navigation. When I try to do what you suggested navigation demands parameters that I think are from the old navigation.

Reply:
Ah right, you just need to pass a start destination as well:
```
navigation<AuthGraph>(startDestination = LoginRoute) {
        composable<LoginRoute> {
        
        }
        ...
    }
```